tests/integration/awslambda/test_lambda.py

```diff
@@ -377,6 +378,99 @@ def test_runtime_introspection_arm(self, lambda_client, create_lambda_function,
         invoke_result = lambda_client.invoke(FunctionName=func_name)
         snapshot.match("invoke_runtime_arm_introspection", invoke_result)
 
+    @pytest.mark.skipif(is_old_provider(), reason="unsupported in old provider")
+    @pytest.mark.skipif(
+        not use_docker(),
+        reason="Monkeypatching of Docker-related flag not applicable if run locally",
+    )
+    @pytest.mark.only_localstack
+    def test_ignore_architecture(
+        self, lambda_client, create_lambda_function, snapshot, monkeypatch
+    ):
+        """Test configuration to ignore lambda architecture by creating a lambda with non-native architecture."""
+        monkeypatch.setattr(config, "LAMBDA_IGNORE_ARCHITECTURE", True)
+
+        # Assumes that LocalStack runs on native Docker host architecture
+        # This assumption could be violated when using remote Lambda executors
+        native_arch = platform.get_arch()
+        non_native_architecture = (
+            Architecture.x86_64 if native_arch == "arm64" else Architecture.arm64
+        )
+        func_name = f"test_lambda_arch_{short_uid()}"
+        create_lambda_function(
+            func_name=func_name,
+            handler_file=TEST_LAMBDA_INTROSPECT_PYTHON,
+            runtime=Runtime.python3_9,
+            Architectures=[non_native_architecture],
+        )
+
+        invoke_result = lambda_client.invoke(FunctionName=func_name)
+        payload = json.loads(to_str(invoke_result["Payload"].read()))
+        lambda_arch = standardized_arch(payload.get("platform_machine"))
+        assert lambda_arch == native_arch
+
+    @pytest.mark.skipif(is_old_provider(), reason="unsupported in old provider")
+    @pytest.mark.skipif(
+        not is_arm_compatible() and not is_aws(),
+        reason="ARM architecture not supported on this host",
+    )
```

Comment:
I think this is fine for now, but emulation is thinkable in both ways, also arm64 on amd64.

Reply:
interesting point. That could even be added to `is_arm_compatible()` if it becomes relevant in the future.